### DISCUSSION TITLE
Apache2 not reading the index.php file inside the public folder.

### DISCUSSION THREAD

**Discussion by samnitt** [opensourcepos/opensourcepos#4059] [Q&A]
I am trying to run ospos in LAMP ubuntu 24.04 (digital ocean).

Apache 2 is not able to read the index.php file in the public folder. If i rename it to index.html the content of the php file is being displayed. 

what am i missing ?? Please assist

**Comment by jekkos** [comment]
Do you have file permissions set correctly on your extracted archive? The index.php should be sert in such a way that apache can read it.

**Comment by jekkos** [comment]
should be 0644 in this case

**Reply by samnitt** [reply]
Hi thanks, the index.php is extracted correctly and is inside the public folder.

I changed the permission, still getting the same error.

I am not able to install php-openssl , but it is inbuilt in LAMP For Ubunty22.04. Could this be the reason ??

I also tired by downgrading the php to 7.4 version. But again I was not able to install php-openssl

**Reply by samnitt** [reply]
#update - I tired having a test.php

sudo nano /var/www/html/public/test.php

<?php
phpinfo();
?>

The test.php file is loading. Apache2 is able to read the .php extensions

**Reply by samnitt** [reply]
![Screenshot 2024-09-20 at 12 42 28 AM](https://github.com/user-attachments/assets/4a22112d-6e12-4c6e-9452-ea98cb2be003)

I am getting these error when i inserted a code to display error in the index.php file.

**Reply by samnitt** [reply]
![image](https://github.com/user-attachments/assets/8d826d81-aa51-4a1f-96a5-f4465d46ddbb)
 error log file
